app.js

```diff
@@ -1,49 +1,106 @@
 const express = require("express");
 const logger = require("morgan");
+const mongoose = require("mongoose");
 
 const app = express();
+const Recipe = require("./models/Recipe.model");
 
 // MIDDLEWARE
 app.use(logger("dev"));
 app.use(express.static("public"));
 app.use(express.json());
 
-
 // Iteration 1 - Connect to MongoDB
 // DATABASE CONNECTION
+const MONGODB_URI = "mongodb://127.0.0.1:27017/express-mongoose-recipes-dev";
 
-
+mongoose
+  .connect(MONGODB_URI)
+  .then((x) =>
+    console.log(`Connected to Mongo! Database name: "${x.connections[0].name}"`)
+  )
+  .catch((err) => console.error("Error connecting to mongo", err));
 
 // ROUTES
-//  GET  / route
-app.get('/', (req, res) => {
-    res.send("<h1>Lab | Express Mongoose Recipes</h1>");
+//  GET  / route - This is just an example route
+app.get("/", (req, res) => {
+  res.send("<h1>LAB | Express Mongoose Recipes</h1>");
 });
 
-
-//  Iteration 4 - Create recipe route
+//  Iteration 3 - Create a Recipe route
 //  POST  /recipes route
 
+app.post("/recipes", (req, res) => {
+  console.log(req.body);
+  Recipe.create({
+    title: req.body.title,
+    instructions: req.body.instructions,
+    level: req.body.level,
+    ingredients: req.body.ingredients,
+    image: req.body.image,
+    duration: req.body.duration,
+    isArchived: req.body.isArchived,
+    created: req.body.created,
+  })
+    .then((createdRecipe) => {
+      res.json(createdRecipe);
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+});
 
-//  Iteration 5 - Read all recipes
+//  Iteration 4 - Get All Recipes
 //  GET  /recipes route
+app.get("/recipes", (req, res) => {
+  Recipe.find()
+    .then((allRecipes) => {
+      res.json(allRecipes);
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+});
 
-
-//  Iteration 6 - Read a single recipe
+//  Iteration 5 - Get a Single Recipe
 //  GET  /recipes/:id route
 
+app.get("/recipes/:id", (req, res)=>{
+    Recipe.findById(req.params.id)
```

Comment:
also applys for the id in the params 
```suggestion
const { id } = req.params
Recipe.findById(id)
```